kubejs/server_scripts/enigmatica/kubejs/base/recipetypes/thermal/machine/insolator_farmersdelight_crops.js

```diff
@@ -0,0 +1,58 @@
+events.listen('recipes', (event) => {
+    var data = {
+        recipes: [
+            {
+                input: 'farmersdelight:cabbage',
+                outputs: ['farmersdelight:cabbage', 'farmersdelight:cabbage_seeds'],
+				chances: [2.0, 1.1],
+				energy_mod: 1.0,
+				water_mod: 0.5
+            },
+			{
+                input: 'farmersdelight:tomato',
+                outputs: ['farmersdelight:tomato', 'farmersdelight:tomato_seeds'],
+				chances: [2.0, 1.1],
+				energy_mod: 1.0,
+				water_mod: 0.5
+            },
+			{
+                input: 'farmersdelight:onion',
+                outputs: ['simplefarming:onion'],
+				chances: [2.5],
+				energy_mod: 1.0,
+				water_mod: 1.0
+            },
+			{
+                input: 'farmersdelight:rice_panicle',
+                outputs: ['simplefarming:rice_panicle'],
+				chances: [2.5],
+				energy_mod: 1.0,
+				water_mod: 1.5
+            }
+        ]
+    };
+    data.recipes.forEach((recipe) => {
+        event.recipes.thermal.insolator({
+		  type: 'thermal:insolator',
+		  ingredient: {
+			item: recipe.input
+		  },
+		  result: [
+			{
+			  item: recipe.outputs[0],
+			  chance: recipe.chances[0]
+			},
+			{
+			  item: recipe.outputs[1],
+			  chance: recipe.chances[1]
+			},
+			{
```

Comment:
Does this script work? I see you're accessing indexes in the arrays that don't exist.

Reply:
Woops, placed the comment incorrectly :P do all the insolator recipes work?

Reply:
It does work, though I don't know if this is bad practice. It seems to revert to minecraft:air in the absence of a valid item, and 0 in the absence of a valid float. 

I was a little surprised by it too, so I double checked that all of the recipes were showing up in JEI.

Reply:
Huh alright, if it works it works :D